Formula/opencascade.rb

```diff
@@ -0,0 +1,91 @@
+class Opencascade < Formula
+  desc "3D modeling and numerical simulation software for CAD/CAM/CAE"
+  homepage "https://dev.opencascade.org/"
+  url "https://github.com/FreeCAD/homebrew-freecad/releases/download/0/opencascade-7.2.0.tgz"
+  sha256 "f9a81ca2267635773af9fff80625735dd4f65389af986fe93a94263081b422e9"
+
+  bottle do
+    cellar :any
+    sha256 "e817f05ac747235d8453a2d15fb076f90f87b30a872ff6944cc5487f05058cbe" => :sierra
+    sha256 "9be44acf5036801dda4981b48693312f7cba3daffa73e80c50ea207a9da7407e" => :el_capitan
+    sha256 "790964a5e17e0021228f03b32bdabdc1657aa89aa56f668d8b998ec5f1ce42bc" => :yosemite
+    sha256 "5ef4d49225c00242f5ab1cc04804f3d4b931c2fc82713c4a9457e57c50f94c33" => :x86_64_linux
+  end
+
+  option "without-opencl", "Build without OpenCL support"
+  option "with-extras", "Install documentation (~17 MB), source files (~113 MB), samples and templates"
+  option "with-test", "Install tests (~55MB)"
+  deprecated_option "with-tests" => "with-test"
+
+  depends_on "cmake" => :build
+  depends_on "freetype"
+  depends_on "freeimage" => :recommended
```

Comment:
Let's make `freeimage`, `tbb` and `gl2ps` mandatory.
Well remove the `test` option, as well as `without-opencl`.
For extras, we want to remove that option, I'd lean toward including docs (and thus build dependency on `doxygen`), but not source and the rest.

Reply:
All great feedback however I don't think we should include the docs or extras because it increases the bottle size by nearly 1GB which wastes significant recurring resources on Travis and CircleCI (`opencascade` is a library/SDK).  I will do the following:

1. add the mandatory for `freeimage`, `tbb`, `gl2ps`
2. remove `without-opencl`
3. leave `test` and `extras` for now.  I will try to do some analysis but on bottle-size but it was a significant hit in the past and the Travis team has been reaching out to macOS users with continuous integration builds asking them to optimize.  For `FreeCAD`, I developed a [Homebrew ports cache](https://github.com/FreeCAD/FreeCAD-ports-cache) to reduce builds times well-beyond what bottled-installs achieve because the homebrew port installs alone were exceeding the standard Travis CI build time allocation.

Reply:
removed `test` because the legacy hb/science formula had deprecated the `test` option and opencascade appears as a "new" formula in hb/core and the build-bot rejects new formula with deprecated options (makes sense)